platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/configuration/DaemonParameters.java

```diff
@@ -150,8 +151,8 @@ public void setRequestedJvmBasedOnJavaHome(@Nullable JavaInfo requestedJvmBasedO
         this.requestedJvmBasedOnJavaHome = requestedJvmBasedOnJavaHome;
     }
 
-    public void applyDefaultsFor(JavaVersion javaVersion) {
-        if (javaVersion.compareTo(JavaVersion.VERSION_1_9) >= 0) {
+    public void applyDefaultsFor(JavaLanguageVersion javaVersion) {
+        if (javaVersion.asInt() >= 9) {
```

Comment:
💭 I wonder if this should be `JavaLanguageVersion.hasJpmsSupport()`?  I feel like we should not resort to reasoning about these sorts of things as numbers whenever we can avoid it.  Instead we should have a pattern of being able to query the language version for its capabilities.  For instance, I could see us wanting to add other queries in the future, e.g. `hasRecordSupport()` or `hasSealedClassesSupport()`.

Reply:
I think so, but as this targets `release` I don't want to add more new APIs than I need to. I made https://github.com/gradle/gradle/issues/29278 to follow-up later.